IceCreamShop.playground/Contents.swift

```diff
@@ -1 +1,91 @@
+import UIKit
 
+struct Flavor {
+    var name: String
+    var rating: Double
+}
+
+enum Size: Double {
+    case small = 1.00
+    case medium = 2.88
+    case large = 5.25
+}
+
+struct Cone {
+    let flavor: Flavor
+    let topping: String?
+    let size: Size
+    
+    func eat() {
+        print("Yum, I love \(flavor.name)!)")
+    }
+}
+
+class IceCreamShop {
+    var flavors: [Flavor]
+    var sizes: [Size]
+    var toppings: [String?]
+    var totalSales: Double = 0.0
+
+    init(flavors: [Flavor], sizes: [Size], toppings: [String?]) {
+        self.flavors = flavors
+        self.sizes = sizes
+        self.toppings = toppings
+    }
+//    func listTopFlavors() {
+//        var topFlavors: [String] = []
+//
+//        for flavor in flavors {
+//            if flavor.rating >= 4.0 {
+//                topFlavors.append(flavor.name)
+//                }
+//            print("Our top flavors are \(flavor.name)")
+//            }
+//        }
+        
+    func listTopFlavors(flavors: [Flavor]) {
+
+        var topFlavors: [String] = []
+
+        for flavor in flavors {
+            if flavor.rating >= 4.0 {
+                topFlavors.append(flavor.name)
+                print("Our top flavors are \(flavor.name)!")
+            }
+        }
+    }
+
+    func orderCone(flavors: Flavor, toppings: String?, sizes: Size) -> Cone? {
+        
+        let myCone = Cone(flavor: flavors, topping: toppings ?? " ", size: sizes)
+ 
+        self.totalSales += myCone.size.rawValue
+        
+        if myCone.topping != nil {
+            print("Your \(myCone.flavor) with \(String(describing: myCone.topping)) is \(myCone.size)")
+            } else {
+                print("Your \(myCone.flavor) is \(myCone.size)")
+                }
```

Comment:
Make sure you use control + i to re-indent your code properly. Also try to make sure when you are printing values you are getting the actual string, so in this case the .name property on flavor. Avoid using String(describing:) with optionals. Instead consider using the nil coalescing operator. All those fixes together would look like this:

``` swift
if myCone.topping != nil {
    print("Your \(myCone.flavor.name) with \(myCone.topping ?? "") is \(myCone.size)")
} else {
    print("Your \(myCone.flavor.name) is \(myCone.size)")
}
```